### PR TITLE
[AutoDiff] Add negative test for SR-12641.

### PR DESCRIPTION
**Pull request by dan-zheng**
Add negative test exposing SR-12641: SILGen verification error regarding
reabstraction thunking for `@differentiable` functions.

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please smoke test

File: test/AutoDiff/compiler_crashers/sr12641-silgen-immutable-address-use-verification-failure.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// RUN: not --crash %target-swift-frontend -emit-sil %s
+
```
Comment (marcrasi) [inline test/AutoDiff/compiler_crashers/sr12641-silgen-immutable-address-use-verification-failure.swift:2]: Needs `// REQUIRES: asserts` ?

**Review by marcrasi — APPROVED** [review]

File: test/AutoDiff/compiler_crashers/sr12641-silgen-immutable-address-use-verification-failure.swift (hunk 1 above)
Comment (dan-zheng) [reply test/AutoDiff/compiler_crashers/sr12641-silgen-immutable-address-use-verification-failure.swift:2]: Yes, thanks! Done.

**Comment by dan-zheng** [comment]
@swift-ci Please smoke test

**Pull request merged**